src/RichText.webmodeler.ts

```diff
@@ -1,55 +1,26 @@
 import { Component, createElement } from "react";
 
 import { RichText, RichTextProps } from "./components/RichText";
-import { RichTextContainerProps } from "./components/RichTextContainer";
+import RichTextContainer, { RichTextContainerProps } from "./components/RichTextContainer";
 
-import { parseStyle } from "./utils/ContainerUtils";
 import { ValidateConfigs } from "./components/ValidateConfigs";
 
-interface PreviewState {
-    reloadEditor: boolean;
-}
-
 // tslint:disable-next-line class-name
-export class preview extends Component<RichTextContainerProps, PreviewState> {
-    constructor(props: RichTextContainerProps) {
-        super(props);
-
-        this.state = {
-            reloadEditor: false
-        };
-    }
-
+export class preview extends Component<RichTextContainerProps, {}> {
     render() {
-        return !this.state.reloadEditor
-            ? createElement(ValidateConfigs, { ...this.props as RichTextContainerProps, showOnError: true },
-                createElement(RichText, preview.transformProps(this.props)))
-            : null;
-    }
-
-    componentWillReceiveProps() {
-        this.setState({ reloadEditor: true });
-    }
-
-    componentDidUpdate() {
-        if (this.state.reloadEditor) {
-            this.setState({ reloadEditor: false });
-        }
+        return createElement(ValidateConfigs, { ...this.props as RichTextContainerProps, showOnError: true },
+            createElement(RichText, preview.transformProps(this.props))
+        );
     }
 
     private static transformProps(props: RichTextContainerProps): RichTextProps {
         const valueAttribute = props.stringAttribute ? props.stringAttribute.split(".")[ 2 ] : "";
 
         return {
-            className: props.class,
-            customOptions: props.customOptions,
-            editorOption: props.editorOption,
-            maxNumberOfLines: props.maxNumberOfLines,
-            minNumberOfLines: props.minNumberOfLines,
+            ... RichTextContainer.getPartialRichTextProps(props) as RichTextProps,
```

Comment:
Personally, no so fan of this solution. as you need to move the an other file to see which props are copied.

Reply:
Inlined them all. Same in the container.